### PR TITLE
Fix Pre-release Regression: Correct JSON Web Key (JWK) Encoding for RSA Public Key Export

### LINKED ISSUE #4881: [Bug] Incorrectly Encoded JWK Format in RSA Public Key Export
### Library version used

pre release

### .NET version

A recent pre-release regression was introduced in the RSA public key export function, leading to improperly encoded JSON Web Key (JWK) parameters.

### Scenario

ConfidentialClient - service to service (AcquireTokenForClient)

### Is this a new or an existing app?

This is a new app or experiment

### Issue description and reproduction steps

- Export an in memory key.
- Observe the generated JWK string.
- Attempt to use the JWK in cryptographic operations or with external systems expecting a correctly formatted JWK.

### Relevant code snippets

_No response_

### Expected behavior

correct JWK expected that matches that one used in token validation

### Identity provider

Microsoft Entra ID (Work and School accounts and Personal Microsoft accounts)

### Regression

pre-release

### Solution and workarounds

_No response_

### PR DESCRIPTION
**Pull request by trwalke**
Fixes #4881
This pull request addresses a regression identified in the pre-release phase that caused incorrect encoding of JSON Web Key (JWK) parameters in the RSA public key export function. 

**Changes proposed in this request**
- revert the change.
- Added unit tests to verify the correct encoding of JWK parameters.

**Testing**
unit tests

**Performance impact**
none

**Documentation**
- [ ] All relevant documentation is updated.

### CODE REVIEW

**Comment by fsheik** [comment]
@trwalke - Please add a unit test in PopTests.cs

``` csharp
        [TestMethod]
        public void ValidateCanonicalJwkFormat()
        {
            // Arrange
            var provider = PoPProviderFactory.GetOrCreateProvider();
            var actualCanonicaljwk = provider.CannonicalPublicKeyJwk;

            // Act and Assert

            // Parse the JWK to get the RSA parameters so that we can create a new canonical JWK in expected format
            var jsonWebKey = JsonWebKey.Create(actualCanonicaljwk);
            var expectedCanonicalJwk = $@"{{""{JsonWebKeyParameterNames.E}"":""{jsonWebKey.E}"",""{JsonWebKeyParameterNames.Kty}"":""{JsonWebAlgorithmsKeyTypes.RSA}"",""{JsonWebKeyParameterNames.N}"":""{jsonWebKey.N}""}}";

            Assert.AreEqual(expectedCanonicalJwk, actualCanonicaljwk);
        }
```

File: src/client/Microsoft.Identity.Client/AuthScheme/PoP/InMemoryCryptoProvider.cs [hunk 1]
```diff
@@ -56,9 +56,8 @@ public byte[] Sign(byte[] payload)
         /// </summary>
         private static string ComputeCanonicalJwk(RSAParameters rsaPublicKey)
         {
-            return $@"{{""{JsonWebKeyParameterNames.E}"":""{Base64UrlHelpers.Encode(rsaPublicKey.Exponent)}"",
-               ""{JsonWebKeyParameterNames.Kty}"":""{JsonWebAlgorithmsKeyTypes.RSA}"",
-               ""{JsonWebKeyParameterNames.N}"":""{Base64UrlHelpers.Encode(rsaPublicKey.Modulus)}""}}";
+            //Important: This format cannot be modefied as it needs to be the same as what is used in the service when calculating hashes.
```
Comment (gladjohn) [inline src/client/Microsoft.Identity.Client/AuthScheme/PoP/InMemoryCryptoProvider.cs:59]:
```suggestion
            //Important: This format cannot be modified as it needs to be the same as what is used in the service when calculating hashes.
```

**Review by gladjohn — APPROVED** [review]

**Comment by bgavrilMS** [comment]
Is there a bug tracking this?

**Comment by gladjohn** [comment]
> Is there a bug tracking this?

given that @trwalke is oof, updated with bug

**Review by gladjohn — APPROVED** [review]

**Review by gladjohn — APPROVED** [review]

**Pull request merged**